system/jlib/jarray.hpp

```diff
@@ -440,7 +439,7 @@ class ReferenceArrayMapper
     {
         member = &newValue;
     }
-    static inline void destruct(ITEM * & member)
+    static inline void destruct(ITEM * & member __attribute((unused)))
```

Comment:
would it be worth defining

    #define __unused(x) x __attribute((unused))

and then use

    static inline void destruct(ITEM * & __unused__(member))
or
    static inline void destruct(__unused(ITEM * & member))